domains/transfer/Transfer.API/API/Program.cs

```diff
@@ -86,11 +92,67 @@
 
 builder.Services.AddLogging();
 
+
 builder.Services.AddTransfer(builder.Configuration);
 builder.Services.AddCvr();
 builder.Services.AddConnection();
 builder.Services.AddClaimServices();
 
+builder.Services.AddControllers(options => options.Filters.Add<AuditDotNetFilter>())
+    .AddJsonOptions(options =>
+        options.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter()));
+Configuration.Setup()
+    .UseEntityFramework(ef => ef
+        .AuditTypeExplicitMapper(config => config
+            .Map<TransferAgreement, TransferAgreementHistoryEntry>((evt, eventEntry, historyEntity) =>
+            {
+                var actorId = evt.CustomFields.ContainsKey("ActorId")
+                    ? evt.CustomFields["ActorId"].ToString()
+                    : null;
+                var actorName = evt.CustomFields.ContainsKey("ActorName")
+                    ? evt.CustomFields["ActorName"].ToString()
+                    : null;
+
+                historyEntity.Id = Guid.NewGuid();
+                historyEntity.CreatedAt = DateTimeOffset.UtcNow;
+                historyEntity.AuditAction = eventEntry.Action;
+                historyEntity.ActorId = actorId ?? string.Empty;
+                historyEntity.ActorName = actorName ?? string.Empty;
+
+                switch (eventEntry.Action)
+                {
+                    case "Insert":
+                        historyEntity.TransferAgreementId = (Guid)eventEntry.ColumnValues["Id"];
+                        break;
+                    case "Update":
+                    {
+                        historyEntity.TransferAgreementId = (Guid)eventEntry.PrimaryKey.Values.First();
+                        break;
+                    }
+                }
+
+                return true;
+            })
+            .Map<ClaimSubject, ClaimSubjectHistory>((evt, eventEntry, historyEntity) =>
```

Comment:
This is being done twice? Here and in startup. I would expect it only to be in startup

Reply:
Moved to only program